numpy/lib/tests/test_shape_base.py

```diff
@@ -2,16 +2,78 @@
 
 import numpy as np
 import warnings
+import functools
 
 from numpy.lib.shape_base import (
     apply_along_axis, apply_over_axes, array_split, split, hsplit, dsplit,
-    vsplit, dstack, column_stack, kron, tile, expand_dims,
+    vsplit, dstack, column_stack, kron, tile, expand_dims, take_along_axis
```

Comment:
Needs tests for `put_along_axis` as well...

Reply:
Will add these in a future commit. There's not any obvious way to test the argfunc/func equivalence like there was for `take_along_axis`, as far as I can tell.

Reply:
Fine to do in a separate commit. One can test, e.g., adding a very large number and then ensuring that `argmax` gives back `indices`.